sdk/core/System.ClientModel/tests/client/ModelReaderWriter/Models/DiscriminatorSet/ModelX.cs

```diff
@@ -112,15 +113,16 @@ private void Serialize(Utf8JsonWriter writer, ModelReaderWriterOptions options)
             writer.WriteEndObject();
         }
 
-        internal static ModelX DeserializeModelX(JsonElement element, ModelReaderWriterOptions options = default)
+        internal static ModelX DeserializeModelX(JsonElement element, ModelReaderWriterOptions? options = default)
         {
             options ??= ModelReaderWriterHelper.WireOptions;
 
             if (element.ValueKind == JsonValueKind.Null)
             {
-                return null;
+                throw new JsonException($"Invalid JSON provided to deserialize type '{nameof(ModelX)}'");
             }
-            string kind = default;
+
+            string? kind = default;
```

Comment:
Required properties are declared as nullable and then later checked to make sure they have a value before passing them to the type's constructor.